### PR TITLE
Add a cert checker for CT example based on fixchain

### PR DESCRIPTION
**Pull request by Martin2112**
+ tests. Also moves more code that depends on CT into the correct part of the tree.

### CODE REVIEW

File: examples/ct/cert_checker.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/google/certificate-transparency/go/asn1"
+	"github.com/google/certificate-transparency/go/x509"
+)
+
+// OID of the non-critical extension used to mark pre-certificates, defined in RFC 6962
+var ctPoisonExtensionOid = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 11129, 2, 4, 3}
```
Comment (AlCutter) [inline examples/ct/cert_checker.go]: /Oid/OID/

File: examples/ct/cert_checker.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		KeyUsages:         []x509.ExtKeyUsage{x509.ExtKeyUsageAny}}
+
+	chains, err := chain[0].Verify(verifyOpts)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(chains) == 0 {
+		return nil, errors.New("No path to root found when trying to validate chains")
```
Comment (AlCutter) [inline examples/ct/cert_checker.go]: /No/no/

File: examples/ct/cert_checker.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		for i, certInChain := range chainMinusRoot {
+			if certInChain != chain[i] {
+				continue
+			}
+		}
+
+		return chainMinusRoot, nil
+	}
+
+	return nil, errors.New("No RFC compliant path to root found when trying to validate chain")
```
Comment (AlCutter) [inline examples/ct/cert_checker.go]: /No/no/

File: examples/ct/cert_checker.go (hunk 1 above)
Comment (Martin2112) [inline examples/ct/cert_checker.go]: Done

File: examples/ct/cert_checker.go (hunk 2 above)
Comment (Martin2112) [inline examples/ct/cert_checker.go]: Done

File: examples/ct/cert_checker.go (hunk 3 above)
Comment (Martin2112) [inline examples/ct/cert_checker.go]: Done.

**Pull request merged**